### PR TITLE
PMM-7508 Add get cluster expected resources

### PR DESCRIPTION
**Pull request by tiagomotasantos**
[PMM-7508](https://jira.percona.com/browse/PMM-7508)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1591

- [ ] Links to other linked pull requests (optional).
   - https://github.com/percona/pmm/pull/704
   - https://github.com/Percona-Lab/pmm-api-tests/pull/204
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: services/management/dbaas/xtradb_cluster_service.go [hunk 1]
```diff
@@ -287,6 +287,32 @@ func (s XtraDBClusterService) RestartXtraDBCluster(ctx context.Context, req *dba
...
+		return nil, err
+	}
+
+	clusterSize := int64(req.Params.ClusterSize)
+	memory := (req.Params.Pxc.ComputeResources.MemoryBytes + req.Params.Proxysql.ComputeResources.MemoryBytes) * clusterSize
+	cpu := int64(req.Params.Pxc.ComputeResources.CpuM+req.Params.Proxysql.ComputeResources.CpuM) * clusterSize
+	disk := (req.Params.Pxc.DiskSize + req.Params.Proxysql.DiskSize) * clusterSize
+
+	if settings.PMMPublicAddress != "" {
+		memory += 1000000000 * clusterSize
```
Comment (jprukner) [inline services/management/dbaas/xtradb_cluster_service.go:303]: @tiagomotasantos I don't understand why is it 1G? 
`kubectl describe pod newclustername-pxc-2`
```
Containers:
  pmm-client:
    Container ID:   docker://512e26e1b24c8eed9f33df532c8fc1e1710d694c658d87255af045c84678684d
    Image:          perconalab/pmm-client:dev-latest
    Image ID:       docker-pullable://perconalab/pmm-client@sha256:e6b04f556b0bf9702b2d9164122b9d6416245c7fe245669f0baa31221402302c
    Ports:          7777/TCP, 30100/TCP, 30101/TCP, 30102/TCP, 30103/TCP, 30104/TCP, 30105/TCP
    Host Ports:     0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP
    State:          Running
      Started:      Tue, 09 Mar 2021 11:49:28 +0100
    Last State:     Terminated
      Reason:       Error
      Exit Code:    137
      Started:      Tue, 09 Mar 2021 08:59:23 +0100
      Finished:     Tue, 09 Mar 2021 11:12:26 +0100
    Ready:          True
    Restart Count:  2
    Requests:
      cpu:     500m
      memory:  500M
```

File: services/management/dbaas/psmdb_cluster_service.go [hunk 2]
```diff
@@ -255,6 +255,32 @@ func (s PSMDBClusterService) RestartPSMDBCluster(ctx context.Context, req *dbaas
...
+	settings, err := models.GetSettings(s.db.Querier)
+	if err != nil {
+		return nil, err
+	}
+
+	clusterSize := int64(req.Params.ClusterSize)
+	memory := req.Params.Replicaset.ComputeResources.MemoryBytes * 2 * clusterSize
+	cpu := int64(req.Params.Replicaset.ComputeResources.CpuM) * 2 * clusterSize
+	disk := req.Params.Replicaset.DiskSize*3 + req.Params.Replicaset.DiskSize*clusterSize
+
```
Comment (jprukner) [inline services/management/dbaas/psmdb_cluster_service.go:269]: Maybe it's just my instance but cluster of size 4 **or** 5 creates **five rs pods**. Could please confirm this and if it is so, could you take this into consideration? @tiagomotasantos

**Review by jprukner — CHANGES_REQUESTED** [review]

File: services/management/dbaas/psmdb_cluster_service.go (hunk 2 above)
Comment (tiagomotasantos) [reply services/management/dbaas/psmdb_cluster_service.go:269]: I checked the behaviour on minikube and EKS with different cluster sizes. We create N mongos pods, N rs pods and 3 CFG pods (N being the cluster size that the user asks for). Also this is what we pass on the cluster definition. So if you ask for a cluster with 4 nodes you should get 4 rs pods.
Screenshot 2021-03-09 at 11 59 33

Comment (tiagomotasantos) [reply services/management/dbaas/psmdb_cluster_service.go:269]: I tried creating PSMDB cluster with 4 nodes and got 4 rs pods 🤔

Comment (jprukner) [reply services/management/dbaas/psmdb_cluster_service.go:269]: kubectl get pods
```
example-cfg-0                                        2/2     Running   0          90s
example-cfg-1                                        2/2     Running   0          65s
example-cfg-2                                        2/2     Running   0          48s
example-mongos-69f6687b8f-96nz2                      0/1     Running   0          88s
example-mongos-69f6687b8f-d9bwd                      0/1     Running   0          88s
example-mongos-69f6687b8f-gq7rr                      0/1     Running   0          88s
example-mongos-69f6687b8f-qn94z                      0/1     Running   0          88s
example-rs0-0                                        2/2     Running   0          92s
example-rs0-1                                        2/2     Running   0          72s
example-rs0-2                                        2/2     Running   0          52s
example-rs0-3                                        2/2     Running   0          32s
example-rs0-4                                        1/2     Running   0          15s
```
What version of operators are you running?

Comment (tiagomotasantos) [reply services/management/dbaas/psmdb_cluster_service.go:269]: Operator version is what we have on dbaas-controller: `percona-server-mongodb-operator:1.6.0`

Comment (tiagomotasantos) [reply services/management/dbaas/psmdb_cluster_service.go:269]: Also @jprukner if you create a cluster with 3 nodes do you see 3 rs pods or 4?

Comment (jprukner) [reply services/management/dbaas/psmdb_cluster_service.go:269]: It's 3 rs pods when i create 3 node clusters. Strange.

Comment (jprukner) [reply services/management/dbaas/psmdb_cluster_service.go:269]: Whatever, I assume it's some kind of ef up on my side.

**Comment by jprukner** [comment]
@tiagomotasantos The digest hash does not match, might be miss merge error.

**Comment by tiagomotasantos** [comment]
Fixed @jprukner!

**Pull request merged**